src/parser/datetime.rs

```diff
@@ -191,6 +379,62 @@ pub(crate) fn build_parsed_datetime(
     Ok(pdt)
 }
 
+/// Takes a 'date timezone' 'date time timezone' string and splits
+/// it into 'date {time}' and 'timezone' components
+pub(crate) fn split_timestamp_string(value: &str) -> (&str, &str) {
+    // First we need to see if the string contains " +" or " -" because timestamps can come in a format
+    // YYYY-MM-DD {+|m}<tz> (where the timezone string can have colons)
+    let mut cut = value.find(" +").unwrap_or(0);
+
+    if cut == 0 {
+        cut = value.find(" -").unwrap_or(0);
+    }
+
+    if cut > 0 {
+        let (first, second) = value.split_at(cut);
+        return (first.trim(), second.trim());
+    }
+
+    // If we have a hh:mm:dd component, we need to go past that to see if we can find a tz
+    let colon = value.find(':');
+
+    match colon {
+        Some(i) => {
+            let substring = value.get(i..).unwrap_or("");
```

Comment:
it seems like this would be a bug if it failed to unwrap? When would it be the right result to fall back to just returning an empty string?

Reply:
if it failed to unwrap and became an empty string then if I understand the library / control flow correctly we would return on line 413 the way the code is written

logically I guess the way i was thinking of it is that the variable substring represents the part of the timestamp string that contains the time portion and timezone portion - if theres some error in retrieving that then I want to fall back to processing the whole string as a timestamp string without any timezone (so therefore the return is (value, "") )

Reply:
fixed this using if let as described above